main.py

```diff
@@ -51,85 +62,115 @@ class MechaClient(Client):
-                 f"{message}")
+        LOG.debug(f"trigger! Sender is {user}\t in channel {channel}\twith data"
+                  f"{message}")
         if user == IRC.presence:
             # don't do this and the bot can get into an infinite
             # self-stimulated positive feedback loop.
-            log.debug("received message from myself ignoring!.")
+            LOG.debug("received message from myself ignoring!.")
             return None
 
         if not message.startswith(Commands.prefix):
             # prevent bot from processing commands without the set prefix
-            log.debug(f"Message {message} did not have our command prefix. Ignoring.")
+            LOG.debug(f"Message {message} did not have our command prefix. Ignoring.")
             return None
 
         else:  # await command execution
-            await Commands.trigger(message=message,
-                                   sender=user,
-                                   channel=channel)
+
+            invoking_user: User = await User.from_bot(bot=self, nickname=user)
+
+            try:
+                await Commands.trigger(message=message,
+                                       sender=invoking_user,
+                                       channel=channel)
+            except CommandNotFoundException as ex:
+                LOG.exception(ex)
+
+            except Exception as ex:
+                # generate a uuid to attach to the error (to make it easier to find later)
+                uuid = uuid4()
+                # write a unique marker to the log file for easier searching
+                LOG.error(f"Error during command invocation. marker: {uuid}")
+                # write the actual exception, since it may be useful
+                LOG.exception(ex)
+
+                # and report state to the user
+                await self.message(channel if channel else user, f"An error has occured during "
+                                                                 f"command execution. Please let a "
+                                                                 f"techrat  know!"
+                                                                 f" -reference id "
+                                                                 f"{str(uuid)[:8]}")
 
 
 @Commands.command("ping")
 async def cmd_ping(bot, trigger):
     """
     Pongs a ping. lets see if the bots alive (command decorator testing)
     :param bot: Pydle instance.
-    :param trigger: `Trigger` object for the command call.
+    :param trigger: `Context` object for the command call.
     """
-    log.warning(f"cmd_ping triggered on channel '{trigger.channel}' for user "
+    LOG.warning(f"cmd_ping triggered on channel '{trigger.channel}' for user "
                 f"'{trigger.nickname}'")
     await trigger.reply(f"{trigger.nickname} pong!")
 
 
+@require_permission(permissions.RAT)
+@Commands.command("version", "potato", "ver")
+async def cmd_version(bot, trigger: Context):
```

Comment:
Maybe these should also be moved to the `commands` package?

Reply:
I want at least the version / ping command to remain in main, as an indication that Mechasqueak started up OK

Reply:
I agree with having some kind of ping capability that does not rely on other modules being loaded

Reply:
This still relies on other modules being loaded. The commands and permissions modules to be exact.